test/e2e/e2e.go

```diff
@@ -211,58 +214,176 @@ var _ = ginkgo.Describe("[efs-csi] EFS CSI", func() {
+			framework.RunKubectlOrDie("apply", "-k", "github.com/kubernetes-sigs/aws-efs-csi-driver/deploy/kubernetes/overlays/stable")
+
+			ginkgo.By(fmt.Sprintf("Creating efs pvc & pv"))
+			path := ""
+			pvName := f.Namespace.Name
+			// non-tls
+			pv := makeEFSPvWithoutTLS(pvName, path)
+			pv, pvc := createPvPvcWithPv(f, pv)
+			defer func() { f.ClientSet.CoreV1().PersistentVolumes().Delete(pv.Name, &metav1.DeleteOptions{}) }()
+
+			_, pod := launchPod(f, pv, pvc)
+
+			ginkgo.By("Upgrading to the EFS CSI driver under test")
+			updateDriverTo(daemonSetsClient, efsDriverUnderTest)
+
+			validatePodVolumeAccessible(f, pod)
 		})
+
 	})
 })
 
+func deleteDriverPodOnNode(f *framework.Framework, node *v1.Node) {
+	ginkgo.By(fmt.Sprintf("Getting driver pod on node %q", node.Name))
+	labelSelector := labels.SelectorFromSet(labels.Set{"app": "efs-csi-node"}).String()
+	fieldSelector := fields.SelectorFromSet(fields.Set{"spec.nodeName": node.Name}).String()
+	podList, err := f.ClientSet.CoreV1().Pods("kube-system").List(
+		metav1.ListOptions{
+			LabelSelector: labelSelector,
+			FieldSelector: fieldSelector,
+		})
+	framework.ExpectNoError(err, "getting driver pod")
+	framework.ExpectEqual(len(podList.Items), 1, "expected 1 efs csi node pod but got %d", len(podList.Items))
+	driverPod := podList.Items[0]
+
+	ginkgo.By(fmt.Sprintf("Deleting driver pod %q on node %q", driverPod.Name, node.Name))
+	err = e2epod.DeletePodWithWaitByName(f.ClientSet, driverPod.Name, "kube-system")
+	framework.ExpectNoError(err, "deleting driver pod")
+}
+
+func createPvPvcWithPv(f *framework.Framework, pv *v1.PersistentVolume) (*v1.PersistentVolume, *v1.PersistentVolumeClaim) {
+	pvc, pv := makePVCWithPV(pv, f.Namespace.Name, f.Namespace.Name)
+	pvc, pv, err := createEFSPVPVC(f.ClientSet, f.Namespace.Name, pvc, pv)
+	framework.ExpectNoError(err, "creating efs pvc & pv")
+
+	return pv, pvc
+}
+
+func validatePodVolumeAccessible(f *framework.Framework, pod *v1.Pod) {
+	ginkgo.By(fmt.Sprintf("Execing a write to the pod %q in namespace %q", pod.Name, pod.Namespace))
+	command := fmt.Sprintf("touch /mnt/volume1/%s-%s", f.Namespace.Name, time.Now().Format(time.RFC3339))
+	done := make(chan bool)
+	go func() {
+		defer ginkgo.GinkgoRecover()
+		utils.VerifyExecInPodSucceed(f, pod, command)
+		done <- true
+	}()
+	select {
+	case <-done:
+		framework.Logf("verified exec in pod succeeded")
+	case <-time.After(30 * time.Second):
+		framework.Failf("timed out verifying exec in pod succeeded")
+	}
+}
+
+func launchPod(f *framework.Framework, pv *v1.PersistentVolume, pvc *v1.PersistentVolumeClaim) (*v1.Node, *v1.Pod) {
+	node, err := e2enode.GetRandomReadySchedulableNode(f.ClientSet)
+	framework.ExpectNoError(err, "getting random ready schedulable node")
+	command := fmt.Sprintf("touch /mnt/volume1/%s-%s && trap exit TERM; while true; do sleep 1; done", f.Namespace.Name, time.Now().Format(time.RFC3339))
+
+	isPrivileged := false
+	var nodeSelector map[string]string
```

Comment:
nodeSelector is not set correctly, something got lost in the refactoring.

Reply:
It seems like nodeSelector was never set correctly?

Reply:
Yeah I never set it: https://github.com/kubernetes-sigs/aws-efs-csi-driver/blob/master/test/e2e/e2e.go#L225 so with 3 nodes we had a 1/3 chance of actually running the test as intended.

I wouldn't use nodeSelector anyway ,I would rather just set pod.spec.Nodename